client/app/components/edit-service-modal/edit-service-modal-service.factory.js

```diff
@@ -36,34 +37,48 @@
   function EditServiceModalController(serviceDetails, $state, $modalInstance, CollectionsApi, EventNotifications) {
     var vm = this;
 
-    vm.service = serviceDetails;
+    vm.isPristine = isPristine;
+    vm.reset = reset;
     vm.saveServiceDetails = saveServiceDetails;
-
-    vm.modalData = {
-      'action': 'edit',
-      'resource': {
-        'name': vm.service.name || '',
-        'description': vm.service.description || '',
-      },
-    };
+    vm.serviceName = serviceDetails.name;
+    vm.serviceDescription = serviceDetails.description;
 
     activate();
 
     function activate() {
+      vm.original = angular.copy(serviceDetails);
+    }
+
+    function reset() {
+      vm.serviceName = vm.original.name;
+      vm.serviceDescription = vm.original.description;
     }
 
     function saveServiceDetails() {
-      CollectionsApi.post('services', vm.service.id, {}, vm.modalData).then(saveSuccess, saveFailure);
+      var data = {
+        'action': 'edit',
+        'resource': {
+          'name': vm.serviceName || '',
+          'description': vm.serviceDescription || '',
+        },
+      };
+
+      CollectionsApi.post('services', serviceDetails.id, {}, data).then(saveSuccess, saveFailure);
 
       function saveSuccess() {
         $modalInstance.close();
-        EventNotifications.success(vm.service.name + __(' was edited.'));
+        EventNotifications.success(vm.serviceName + __(' was edited.'));
```

Comment:
Could you alter the i18n string here, since not all languages follow the `Subject` + `Predicate` grammar format.
You could use `sprintf` for this and you'll see some existing examples in the code.
```javascript
EventNotifications.success(sprintf(__("%s was edited."), vm.serviceName));
```